src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+        )
+    end
+    # Set filtered locations as n_locs+1 for consistency with time dependent ranks
+    ranks_store[ranks_store.==0.0] .= length(domain.site_ids)+1
+
+    return ranks_store
+
+end
+function run_site_selection(domain::Domain, 
+    scenarios::DataFrame, 
+    sum_cover::NamedDimsArray, 
+    area_to_seed::Float64, 
+    aggregation_function::Function, 
+    iv_type::Int64;
+    target_seed_sites=nothing, 
+    target_shade_sites=nothing)
+
+    ranks = run_site_selection(domain, 
+        scenarios, 
+        sum_cover, 
+        area_to_seed; 
+        target_seed_sites=target_seed_sites,
+        target_shade_sites=target_shade_sites,
+    )
+
+    return aggregation_function(ranks(; intervention=iv_type))
+
+end
+
+
+"""
+    ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)ranks_to_frequencies(rs::ResultSet, iv_type::String; n_ranks=length(ranks.sites))
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,3,A};n_ranks=length(ranks.sites),agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),) where {D,T,A}
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,2,A};n_ranks=length(ranks.sites),agg_func=nothing) where {D,T,A}
+
+Post-processing function for location ranks output of `run_location_selection()`. Gives the frequency 
+with which each location was selected at each rank across the location selection scenarios.
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_ranks` : number of rankings (defualt is number of locations).
+- `agg_func` : Aggregation function to appy after frequencies are calculated.
+
+# Returns
+Frequency with which each location was selected for each rank.
+"""
+function ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)
+    dn = NamedDims.dimnames(ranks)
+    freq_dims = [n for n in dn if n != :scenarios]
+    dn_subset = vcat(freq_dims, [:ranks])
+    freq_elements = vcat(
+        [1:size(ranks, n) for n in dn if n != :scenarios],
+        [1:size(ranks, :sites)],
+    )
+    mn = ([size(ranks, k) for k in freq_dims]..., size(ranks, :sites))
+
+    rank_frequencies = NamedDimsArray(zeros(mn...); zip(dn_subset, freq_elements)...)
+
+    for rank in 1:n_ranks
+        rank_frequencies[ranks=Int64(rank)] .= sum(ranks .== rank; dims=:scenarios)[scenarios=1]
+    end
+
+    return rank_frequencies
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,3,A};
+    n_ranks=length(ranks.sites),
+    agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),
+) where {D,T,A}
+    return agg_func(ranks_to_frequencies(ranks, n_ranks))
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,2,A};
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
```

Comment:
We don't need the first one with a default `agg_func`, do we?

Reply:
If I left out the default aggfunc the default output would include timesteps. Happy to do this, probablky more consistent, but I'd have to add the check for the aggfunc as in the second case

Reply:
Oh I see, then we should document what the default behaviour is. Please clarify the docstring.

Reply:
Will do. Also should I add an `agg_func` instead of `dims` argument to `summed_inverse_rank()`? I'd have to change the normalisation but it would add flexibility (if desired).